python/tests/integration/arcticdb/test_arctic.py

```diff
@@ -166,6 +166,41 @@ def test_library_management_path_prefix(moto_s3_uri_incl_bucket, boto_client):
         _lib = ac["pytest_test_lib"]
 
 
+@pytest.mark.parametrize("arctic_library", [["use_tombstones", "sync_passive.enabled"]], indirect=True)
+def test_read_meta_batch_with_tombstones(arctic_library):
+    lib = arctic_library
+    lib.write_pickle("sym1", 1, {"meta1": 1}, prune_previous_versions=False)
+    lib.write_pickle("sym1", 1, {"meta1": 2}, prune_previous_versions=False)
+    lib.write_pickle("sym2", 1, {"meta2": 1}, prune_previous_versions=False)
+    lib.write_pickle("sym2", 1, {"meta2": 2}, prune_previous_versions=False)
+    lib.delete("sym1", versions=0)
+    lib.delete("sym2", versions=0)
+    results_dict = lib.read_metadata_batch(["sym1", "sym2", "sym3"])
+    assert results_dict["sym1"].metadata == {"meta1": 2}
+    assert results_dict["sym2"].metadata == {"meta2": 2}
+    assert "sym3" not in results_dict
+    assert lib.read_metadata("sym1").metadata == results_dict["sym1"].metadata
+
+
+@pytest.mark.parametrize(
+    "arctic_library", [["use_tombstones", "prune_previous_version", "sync_passive.enabled"]], indirect=True
+)
+def test_read_meta_batch_with_pruning(arctic_library):
+    lib = arctic_library
+    lib.write_pickle("sym1", 1, {"meta1": 1})
+    lib.write_pickle("sym1", 1, {"meta1": 2})
+    lib.write_pickle("sym1", 3, {"meta1": 3})
+    lib.write_pickle("sym2", 1, {"meta2": 1})
+    lib.write_pickle("sym2", 1, {"meta2": 2})
+    lib.write_pickle("sym2", 3, {"meta2": 3})
+    # v0 and v1 should be pruned by now.
+    results_dict = lib.read_metadata_batch(["sym1", "sym2", "sym3"])
+    assert results_dict["sym1"].metadata == {"meta1": 3}
+    assert results_dict["sym2"].metadata == {"meta2": 3}
+    assert "sym3" not in results_dict
+    assert lib.read_metadata("sym1").metadata == results_dict["sym1"].metadata
```

Comment:
This isn't really testing anything different than the previous test, as you don't try and read the metadata of the pruned versions.
It would be good to have a test for the `as_of` `kwarg` as well.

Reply:
All the tests are copied and adapted  from the same test of the store api: test_batch_read_meta_with_tombstones test the already existing batch_read_metadata function

Reply:
Improved the tests now